controllers/greptimedbcluster/deployers/datanode.go

```diff
@@ -153,6 +257,57 @@ func (d *DatanodeDeployer) deleteStorage(ctx context.Context, cluster *v1alpha1.
 	return nil
 }
 
+// isPodRestart checks if the pod needs to be restarted. For convenience, we only compare the necessary fields.
+// TODO(zyy17): Do we have a easy way to implement this?
+func (d *DatanodeDeployer) isPodRestart(new, old appsv1.StatefulSet) bool {
+	var (
+		newPodTemplate = new.Spec.Template
+		oldPodTemplate = old.Spec.Template
+	)
+
```

Comment:
question：By checking whether old.spec is equal to new.spec, we can determine whether to restart?